File: fabric/src/main/java/com/noxcrew/noxesium/network/NoxesiumPacket.java
package com.noxcrew.noxesium.network;

import net.fabricmc.fabric.api.networking.v1.FabricPacket;

/**
 * The basis for a fabric packet as used by Noxesium, these always contain
 * a version varint which allows different versions of the mod to communicate.
 */
public abstract class NoxesiumPacket implements FabricPacket {

    /** The version of this packet. */
    public final int version;

    /**
     * Creates a new Noxesium packet with the given version.
     *
     * @param version The version of this packet, this is always
     *                the first varint of any Noxesium packet and
     *                allows the contents of packets to change
     *                over time without much issue.
     */
    public NoxesiumPacket(int version) {
        this.version = version;
    }
}

File: paper/legacy/src/main/kotlin/com/noxcrew/noxesium/legacy/api/network/NoxesiumPacket.kt
package com.noxcrew.noxesium.legacy.api.network

/**
 * The basis for a server-side packet as used by Noxesium.
 */
public abstract class NoxesiumPacket(
    /** The packet type of this packet. */
    public open val type: PacketType<*>,
)
